### ISSUE TITLE
customFormFields not working anymore

### ISSUE THREAD

**Issue by andreiciceu** [martndemus/ember-form-for#167]
With the latest build, customFormFields are not working anymore.

**Comment by brandynbennett** [comment]
I placed debuggers inside of https://github.com/martndemus/ember-form-for/blob/master/addon/helpers/form-for/merge-custom-form-field.js and it's not being called at all. Even though I have `customFormFields` inside the `each` loop.

**Comment by hectorh30** [comment]
Same here. Did you find any workaround?

**Comment by andreiciceu** [comment]
Not yet, I reverted to using version *2.0.0-alpha.13*

Know bug in that version, form.focus() generates an error. To avoid this,
in form-for.js line 48, use this
```
schedule('afterRender', () => {
    try {
        this.$().focus()
    }
    catch(e) {
        console.error(e);
    }
});
break;
```

**Comment by andreiciceu** [comment]
@hectorh30 
Not yet, I reverted to using version *2.0.0-alpha.13*

Know bug in that version, form.focus() generates an error. To avoid this,
in form-for.js line 48, use this
```
schedule('afterRender', () => {
    try {
        this.$().focus()
    }
    catch(e) {
        console.error(e);
    }
});
break;
```

**Comment by hectorh30** [comment]
Thanks. I made it work with version 2.0.0-alpha.14, but it required me to go back to ember 2.9.1 (I'm still trying to figure that part out).

Anyways, as far as I could dig this feature was apparently dropped in https://github.com/martndemus/ember-form-for/pull/144/files#diff-20adfe4374a771990fba31f95c8d61b3L20, but I have no idea why.

**Comment by hectorh30** [comment]
Ok, I think this might be related. I quote from the [Ember 2.10 release blog post](https://emberjs.com/blog/2016/11/30/ember-2-10-released.html#toc_other-notable-changes):

> Finally, it has come to our attention that it is quite common for developers to accidentally mutate the arguments passed into a helper. This is problematic as these objects are often shared or reused, causing subtle issues that are very difficult to debug.

> To help track down these kind of issues, these objects are now frozen in the debug build before they are passed into helpers. As a result, you will get an error if you attempt to modify these objects directly. It is recommended that you clone these objects (e.g. using Object.assign and Array#slice) before making any modifications to them.

Apparently it is not a good idea to _mutate the arguments passed into a helper_, which I think was the case here.

Related: https://github.com/emberjs/ember.js/issues/14189

**Comment by martndemus** [comment]
Correct, I removed it, because it stopped working in newer Ember versions.

**Comment by andreiciceu** [comment]
Ah, I see. So what be the right way of adding custom fields now?

**Comment by andreiciceu** [comment]
For anyone interested, the only way I found of adding custom components was by modifying the `addon/templates/components/form-for.hbs` and `addon/templates/components/fields-for.hbs`, inserting components as needed. e.g.:
`
datepicker-field=(component 'form/form-for/datepicker-field' object=object update=update)
`

**Comment by miguelcobain** [comment]
For the record, I managed to work around this by proving an `onChange` action that is used to send the new object out of the helper.

Check this test for more details: https://github.com/miguelcobain/ember-leaflet/blob/master/tests/integration/helpers/ember-leaflet-assign-to-test.js

**Comment by marxsk** [comment]
@miguelcobain I'm waiting for responses for PR about migrating from one-way-controls. When this will be done, there should be a right situation for fixing this issue properly. 

Also, there should be a better way, how to implement your own fields.
